modules/users/server/controllers/users/users.authentication.server.controller.js

```diff
@@ -102,21 +99,53 @@ exports.signup = function(req, res) {
     // Login
     function(user, done) {
       req.login(user, function(err) {
-        if (!err) {
-          // Remove sensitive data befor sending user
-          user = profileHandler.sanitizeProfile(user);
-          return res.json(user);
-        }
-        done(err);
+        // Remove sensitive data befor sending user
+        user = profileHandler.sanitizeProfile(user);
+
+        done(err, user);
       });
     }
 
-  ], function(err) {
+  ], function(err, user) {
+
+    var statsObject = {
+      namespace: 'signup',
+      counts: {
+        count: 1
+      },
+      time: new Date(),
+      tags: {},
+      meta: {}
```

Comment:
The `statsObject` looks good. `time`, `meta` and `tags` properties are optional, so `meta` can be removed and `time` depends on preference (time now vs. time of arrival to stathat).